Mathlib/AlgebraicTopology/SimplicialSet.lean

```diff
@@ -84,6 +84,63 @@ scoped[Simplicial] notation3 "Δ[" n "]" => SSet.standardSimplex.obj (SimplexCat
 instance : Inhabited SSet :=
   ⟨Δ[0]⟩
 
+section Product
+
+/-- `Prod X Y` is the explicit binary product of simplicial sets `X` and `Y`. -/
+def Prod (X Y : SSet) : SSet where
+  obj n := X.obj n × Y.obj n
+  map f a := (X.map f a.1, Y.map f a.2)
+
+/-- The first projection of `Prod X Y`, onto X. -/
+@[simp]
+def Prod.fst {X Y : SSet} : (Prod X Y) ⟶ X where
+  app _ a := a.1
+
+/-- The second projection of `Prod X Y`, onto Y. -/
+@[simp]
+def Prod.snd {X Y : SSet} : (Prod X Y) ⟶ Y where
+  app _ a := a.2
+
+/-- The binary fan whose point is `Prod X Y`. -/
+@[simps!]
+def binaryProductCone (X Y : SSet.{0}) : BinaryFan X Y :=
+  BinaryFan.mk (Prod.fst) (Prod.snd)
+
+@[simp]
+theorem binaryProductCone_fst (X Y : SSet) : (binaryProductCone X Y).fst = Prod.fst :=
+  rfl
+
+@[simp]
+theorem binaryProductCone_snd (X Y : SSet) : (binaryProductCone X Y).snd = Prod.snd :=
+  rfl
+
+/-- The morphism from the point of any binary fan to `Prod X Y`. -/
+@[simp]
+def binaryProductLift {X Y : SSet} (s : BinaryFan X Y) : s.pt ⟶ (binaryProductCone X Y).pt where
+  app n x := ((s.fst).app n x, (s.snd).app n x)
+  naturality j k g := by
+    ext a
+    simp [FunctorToTypes.naturality]
+
+/-- `Prod X Y` is a limit cone. -/
+@[simps]
+def binaryProductLimit (X Y : SSet) : IsLimit (binaryProductCone X Y) where
+  lift (s : BinaryFan X Y) := binaryProductLift s
+  fac _ j := Discrete.recOn j fun j ↦ WalkingPair.casesOn j rfl rfl
+  uniq _ _ ht := by
+    simp only [← ht ⟨WalkingPair.right⟩, ← ht ⟨WalkingPair.left⟩, binaryProductLift]
+    congr
+
+/-- `Prod X Y` is a binary product for `X` and `Y`. -/
+def binaryProductLimitCone (X Y : SSet) : Limits.LimitCone (pair X Y) :=
+  ⟨_, binaryProductLimit X Y⟩
+
+/-- The categorical binary product of simplicial sets is `Prod X Y`. -/
+noncomputable def binaryProductIso (X Y : SSet) : X ⨯ Y ≅ Prod X Y :=
+  limit.isoLimitCone (binaryProductLimitCone X Y)
+
```

Comment:
Some API is missing around `binaryProductIso`. For example, the lemma that says that composing `Prod.fst` with `(binaryProductIso _ _).hom` is the first projection, similarly for the second projection and analogous lemmas for the `(binaryProductIso _ _).inv`.

Reply:
Knew I was forgetting something, thanks!